include/zephyr/net/offloaded_netdev.h

```diff
@@ -29,6 +29,18 @@ extern "C" {
  * @{
  */
 
+/** Types of offloaded netdev L2 */
+enum offloaded_net_if_types {
+	/** Ethernet devices */
+	L2_OFFLOADED_NET_IF_TYPE_ETHERNET	= BIT(0),
+
+	/** Modem */
+	L2_OFFLOADED_NET_IF_TYPE_MODEM		= BIT(1),
+
+	/** IEEE 802.11 Wi-Fi */
+	L2_OFFLOADED_NET_IF_TYPE_WIFI		= BIT(2),
```

Comment:
Shouldn't the device types be mutually exclusive? Not sure I understand why is this defined as a bitfield.

Reply:
Yeah, I was implementing them differently and this was a leftover, will clean it up.